src/DataViews/include/DataViews/FrametrackDataView.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef DATA_VIEWS_FRAMETRACK_DATA_VIEW_H_
+#define DATA_VIEWS_FRAMETRACK_DATA_VIEW_H_
+
+#include <cstdint>
+
+#include "DataViews/DataView.h"
+
+namespace orbit_data_views {
+
+// The abstract class for all the DataView that can enable/disable frametracks
+class FrametrackDataView : public DataView {
```

Comment:
Good point. How about another abstract class? E. g. `HookingDataView` that `FrameTrackDataView` would also inherit from?

Reply:
It's not just about hooking. All actions, or at least the ones shared by at least two concrete `DataView`s, seem to be handled at the top-level. So what you are proposing is the opposite of what I pointed out, as this would go towards a model where all actions that don't apply to all `DataView`s (most of them) are separated into multiple abstract classes. I'm not sure we want that. Also, at that point most `DataView`s would need to inherit from more than one of these classes and that's where we get to the multiple inheritance we don't want.

Reply:
I agree with Davide. I would like to avoid having abstract class for actions. 
Adding the `FrameTrackDataView` makes me feel like that the frame track related actions are treated differently. 
And if we do the same for other actions, that might result in multiple inheritance.

Reply:
> Maybe, we'll stumble upon it in the future. Maybe, we'll even stumble upon diamond inheritance.

Are these maybes? Imagine you extract `OnDisassemblyRequested` to a different intermediate class. You immediately have the diamond problem.

Reply:
> Adding the FrameTrackDataView makes me feel like that the frame track related actions are treated differently.

In a sense, they are: for D vs MS/MA. 
How about `OnJumpToRequested`? It is already implemented in `DataView`, yet redefined in `LiveFunctionsDataView`. Isn't this the seemingly different treatment? 

IMO, if architecture is "good", there should be single definition for every method in the inheritance chain. Right now we redefine implementations. The path I suggest would ultimately allow us to get rid of this. 

From the style guide: "Multiple inheritance is permitted, but multiple implementation inheritance is strongly discouraged". And I don't see how we get stuck with multiple implementation inheritance.

Reply:
> Are these maybes? Imagine you extract OnDisassemblyRequested to a different intermediate class. You immediately have the diamond problem.

Why would I? I might be missing smth, but it could be in its own abstract class, that would, perhaps inherit from another abstract class implementing `GetFunctionInfoFromRow`. I fail to see where we have multiple implementations of `OnDisassemblyRequested`

Reply:
> How about OnJumpToRequested? It is already implemented in DataView, yet redefined in LiveFunctionsDataView.

The default implementation is empty. I assume this is to save every single subclass that is not interested in that action from having to provide an empty implementation. I agree it's not great to have methods that many subclasses don't need. However, this is not as bad as a redefinition, so if we can improve further with a cleaner design, sure, but I don't think it's of primary importance in the discussion.

> From the style guide: "Multiple inheritance is permitted, but multiple implementation inheritance is strongly discouraged". And I don't see how we get stuck with multiple implementation inheritance.

> I might be missing smth, but it could be in its own abstract class, that would, perhaps inherit from another abstract class implementing GetFunctionInfoFromRow.

From the code I see, `FrametrackDataView` inherits from `DataView`. So these classes wouldn't be interfaces.
Multiple inheritance is permitted because that also includes implementing interfaces in C++. Style guide: "Interface inheritance" is inheritance from a pure abstract base class (one with no state or defined methods); all other inheritance is "implementation inheritance". 
If the classes inherit from `DataView` which has member variables, not only is this not interface inheritance, but you have a diamond.

Reply:
> However, this is not as bad as a redefinition

Bit redefining an empty implementation is still a redefinition, no? 

> other inheritance is "implementation inheritance".

So does the standard discourage multiple inheritance from e. g. classes `Foo {void foo() {  } }` and `Bar {void bar() {  } }` ?

Reply:
It was agreed offline, that either we make an effort to refactor the whole DataView, or let it be. This shall for now be fixed with a minimal change.